app/controllers/queue_items_controller.rb

```diff
@@ -0,0 +1,43 @@
+class QueueItemsController < ApplicationController
+  before_action :require_user
+
+  def index
+    @queue_items = current_user.queue_items
+  end
+
+  def create
+    video = Video.find(params[:video_id])
+    queue_video(video)
+    redirect_to my_queue_path
+  end
+
+  def destroy
+    queue_item = QueueItem.find(params[:id])
+    queue_item.delete unless queue_item.user != current_user
+    update_queue_items_positions()
```

Comment:
yea passing arguments is definitely matter of taste, but it depends on what you think is more readable I guess. I recommend reading Eloquent Ruby - you will learn a lot from there also about style.